src/libmongoc/src/mongoc/mongoc-collection.c

```diff
@@ -905,7 +905,7 @@ mongoc_collection_estimated_document_count (
       ret = mongoc_collection_read_command_with_opts (
          coll, &cmd, read_prefs, opts, reply_ptr, error);
 
-      if (error && error->code == MONGOC_ERROR_COLLECTION_DOES_NOT_EXIST) {
+      if (!ret && error->code == MONGOC_ERROR_COLLECTION_DOES_NOT_EXIST) {
```

Comment:
libmongoc functions do not require `error` be initialized. It is valid to do:

```
bson_error_t err; /* uninitialized */
bson_t reply;
bool ret;

ret = mongoc_collection_estimated_document_count (coll, NULL /* opts */, NULL /* read_prefs */, &reply, &err);
```

The boolean return determines whether `error` was set by the called function.
`reply` is always initialized, so callers can always call `bson_destroy` on the `reply` regardless of error.

Reply:
Oh cool, I actually wasn't familiar with the difference between a wild and a null pointer. Surprised this wasn't caught in Valgrind in the C driver...

Reply:
Haha, I haven't heard the term "wild" pointer before, but I like it. I've always referred to it as "uninitialized".